ReactWindows/ReactNative.Shared/Animated/NativeAnimatedNodesManager.cs

```diff
@@ -472,7 +475,7 @@ public void RunUpdates(TimeSpan renderingTime)
                     var animation = _activeAnimations[animationId];
                     if (animation.HasFinished)
                     {
-                        animation.EndCallback.Invoke(new JObject
+                        animation.EndCallback?.Invoke(new JObject
```

Comment:
In what case can EndCallback be null?

Reply:
I suspect if we added an ArgumentNullException here:
https://github.com/reseul/react-native-windows/blob/e7ceea2c62c40b879a88e86a38cc3bb65fc7cee4/ReactWindows/ReactNative.Shared/Animated/AnimationDriver.cs#L17

We'd find that EndCallback is never null.

Reply:
Well, EndCallback is null when using this new tracking animation. Not sure how it works and I didn't spend time, but RNTester was choking there.  I added the '?' after verifying that the Android code had it as well.
In fact to be honest, when I started this work I didn't even know what this new animation was supposed to do. I just ported the Android code to the point it started to work.



---
In reply to: [206494998](https://github.com/Microsoft/react-native-windows/pull/1918#discussion_r206494998) [](ancestors = 206494998)

Reply:
I'll have to pull to try it out - I can't see how the ICallback instance can ever be null, as we never invoke a native method with a null ICallback...

Reply:
It's in the new code (TrackingAnimatedNode.cs):
_manager.StartAnimatingNode(_animationId, _valueNode, _animationConfig, null);

---
In reply to: [206734656](https://github.com/Microsoft/react-native-windows/pull/1918#discussion_r206734656) [](ancestors = 206734656)

Reply:
@reseul - I think there are a few other spots you need to add null coalescing method calls for end callbacks then.

Reply:
Added the req'd changes.